packages/nocodb/src/services/attachments.service.ts

```diff
@@ -216,17 +228,30 @@ export class AttachmentsService {
           } = {};
 
           if (mimeType.includes('image')) {
+            let sharp: typeof Sharp;
             try {
-              const metadata = await sharp(file, {
-                limitInputPixels: true,
-              }).metadata();
+              sharp = (await import('sharp')).default;
+            } catch {
+              // ignore
+            }
 
-              if (metadata.width && metadata.height) {
-                tempMetadata.width = metadata.width;
-                tempMetadata.height = metadata.height;
+            if (sharp) {
+              try {
+                const metadata = await sharp(file, {
+                  limitInputPixels: true,
+                }).metadata();
+
+                if (metadata.width && metadata.height) {
+                  tempMetadata.width = metadata.width;
+                  tempMetadata.height = metadata.height;
+                }
+              } catch (e) {
+                this.logger.error(`${file.path} is not an image file`);
               }
-            } catch (e) {
-              this.logger.error(`${file.path} is not an image file`);
+            } else {
+              this.logger.warn(
+                `Thumbnail generation is not supported in this platform at the moment.`,
+              );
```

Comment:
you can move this to catch block to keep it same with others or vice-versa

Reply:
will move outside in both place